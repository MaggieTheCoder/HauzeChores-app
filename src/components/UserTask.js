import React from "react";
import propTypes from "prop-types";
import completeATask from "../requests/completeATask";
import "../styles/UserTask.css";

const UserTask = ({ taskname, id, userId, setAddedANewTask }) => {
  return (
    <div className="usertask">
      <button
        className="usertask__button"
        type="button"
        onClick={() => {
          completeATask(id, userId, setAddedANewTask);
        }}
      >
        Done!
      </button>
      <p className="usertask__name">{taskname}</p>
    </div>
  );
};

export default UserTask;

UserTask.propTypes = {
  taskname: propTypes.string.isRequired,
  id: propTypes.number.isRequired,
  userId: propTypes.number.isRequired,
  setAddedANewTask: propTypes.func.isRequired,
};
