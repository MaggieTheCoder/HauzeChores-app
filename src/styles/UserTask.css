.usertask-container {
  background-color: #c60982;
  height: 50px;
  padding-top: 5px;
  padding-bottom: 30px;
}

.usertask {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-size: 20px;
  margin-left: 10px;
  padding-left: 100px;
  padding-right: 100px;
}

.usertask__button {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  background-color: #360866;
  padding: 3px 50px 3px 50px;
  border-radius: 15px;
  color: white;
}

.usertask__name {
  color: white;
  font-size: 25px;
}

@media screen and (max-width: 390px) {
  .usertask__button {
    font-size: 20px;
    padding: 1px 5px 1px 5px;
    margin-right: 40px;
  }
  .usertask__name {
    font-size: 15px;
  }
  .usertask {
    justify-content: flex-start;
    padding: 0;
  }
}
