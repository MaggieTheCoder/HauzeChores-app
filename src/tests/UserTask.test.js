import React from "react";
import { render } from "@testing-library/react";
import UserTask from "../components/UserTask";

test("renders UserTask component correctly", () => {
  const { asFragment } = render(<UserTask />);

  expect(asFragment()).toMatchSnapshot();
});
