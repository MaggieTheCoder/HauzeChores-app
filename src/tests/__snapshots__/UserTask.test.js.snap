// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserTask component correctly 1`] = `
<DocumentFragment>
  <div
    class="usertask"
  >
    <button
      class="usertask__button"
      type="button"
    >
      Done!
    </button>
    <p
      class="usertask__name"
    />
  </div>
</DocumentFragment>
`;
